tentative: [pnpm] initial commit

thypon opened the pull request:
<!-- Add brave-browser issue below that this PR will resolve -->
Resolves

<!-- CI-related labels that can be applied to this PR:
* CI/disable-pipeline-step-cache - instruct CI to not cache build steps between runs for the same commit hash
* CI/enable-coverage - enable coverage reporting for your code changes
* CI/enable-test-only-affected - instruct CI to only run tests affected by your change
* CI/run-audit-deps (1) - check for known npm/cargo vulnerabilities (audit_deps)
* CI/run-network-audit (1) - run network-audit
* CI/run-perf-smoke-tests - run smoke performance tests
* CI/storybook-url (1) - deploy storybook and provide a unique URL for each build
* CI/run-upstream-tests - run Chromium unit and browser tests on Linux and Windows (otherwise only on Linux)
* CI/skip-upstream-tests - do not run Chromium unit, or browser tests (otherwise only on Linux)
* CI/run-teamcity - run TeamCity
* CI/skip-teamcity - skip TeamCity
* CI/skip - do not run CI builds (except noplatform)
* CI/run-linux-arm64, CI/run-macos-x64, CI/run-windows-arm64, CI/run-windows-x86 - run builds that would otherwise be skipped
* CI/skip-linux-x64, CI/skip-android, CI/skip-macos-arm64, CI/skip-ios, CI/skip-windows-x64 - skip CI builds for specific platforms

(1) applied automatically when some files are changed (see: https://github.com/brave/brave-core/blob/master/.github/labeler.yml)
-->

<!--
## Checklist:

- Review design docs
  [Browser design principles](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/chrome_browser_design_principles.md)
  [Style guide](https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++.md)
  [Core principles](https://www.chromium.org/developers/core-principles/)
- Ensure there are (tests)[https://www.chromium.org/developers/testing/]. Unit test as much as possible (including edge cases), but also include browser tests covering high level functionality.
- Ensure that there are comments explaining what classes/methods are/do. The "why" is often more important than the "what" in comments. Also update any relevant docs (moving docs from wiki to brave-core if necessary).
- Request security or other review (third-party libraries, rust code, etc...) if applicable [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) [other review](https://github.com/brave/reviews/issues/new/choose)
  Also see [adding third-party libraries](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/adding_to_third_party.md) for general guidelines on using third party code
- Maks sure there is a [ticket](https://github.com/brave/brave-browser/issues) for your issue
- Use Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- Write a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- Squash any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- Add appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- Run `git rebase master` (if needed)
-->

thypon committed:
[pnpm] initial commit

brave-builds commented:
> [!WARNING]
> You have got a presubmit warning. Please address it if possible.

```
Expected one of license headers in existing files:
// Copyright (c) <year> The Brave Authors. All rights reserved.
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this file,
// You can obtain one at https://mozilla.org/MPL/2.0/.

/* Copyright (c) <year> The Brave Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at https://mozilla.org/MPL/2.0/. */

# Copyright (c) <year> The Brave Authors. All rights reserved.
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this file,
# You can obtain one at https://mozilla.org/MPL/2.0/.

<!-- Copyright (c) <year> The Brave Authors. All rights reserved.
     This Source Code Form is subject to the terms of the Mozilla Public
     License, v. 2.0. If a copy of the MPL was not distributed with this file,
     You can obtain one at https://mozilla.org/MPL/2.0/. -->

```

Items:
```
script/generate-tsconfig.py
script/web_discovery_project.py
```

<!-- presubmit-hash=f444685c2f6fa938571e3a41edb7934f82aebf856d89e3fecded62cdd93878c1 -->

thypon committed:
update pnpm lockfile

thypon committed:
jest.config.js: fix transformIgnorePatterns for pnpm

thypon committed:
update pnpm lock

thypon committed:
jest.config.js: simplify transformIgnorePatterns for pnpm

Previous attempt at fixing transformIgnorePatterns tried to list specific
packages, but pnpm's node_modules structure with ES modules requires
transforming most packages. Simplified by only excluding packages that
cause issues (jest, babel, lodash).

Fixes Jest test failures with:
- react-markdown and react-syntax-highlighter ES modules
- bs58/base-x module loading
- All unified ecosystem packages (remark, rehype, hast, mdast, etc.)

thypon committed:
jest.config.js: exclude untracked leo directory from tests

The leo directory is not part of the Brave core repository and requires
build artifacts that don't exist. Excluding it from testPathIgnorePatterns
prevents test failures.

thypon committed:
script/web_discovery_project.py: use npm install with --no-save --yes

npm ci recreates node_modules from scratch on each sync, which is
unnecessary. npm install is faster for incremental updates.

Add --no-save to prevent modifying package-lock.json during sync.
Add --yes to automatically answer yes to any prompts.

thypon committed:
update pnpm lockfile after rebase onto master

thypon committed:
fix(test): add --no-ext-diff to gitPatcher test's getPatch helper

External diff tools (e.g. difftastic) intercept git diff and produce
non-unified output that git apply cannot parse, causing 6 test failures.

atuchin-m commented on script/audit_deps.py:
```diff
@@ -85,24 +85,26 @@ def audit_path(path, args):
+    pnpm_cmd = 'pnpm'
     if sys.platform.startswith('win'):
-        npm_cmd = 'npm.cmd'
+        pnpm_cmd = 'pnpm.cmd'
```
this could also be `pnpm.exe`
https://github.com/brave/brave-core/pull/35603

atuchin-m commented on .prettierignore:
```diff
@@ -38,6 +38,8 @@ components/ai_chat/core/browser/tools/tools.md
 chromium_presubmit_config.json5
 jest.config.js
 snapcraft.yaml
+.eslintrc.js
```
is it intentional?

wknapik commented:
Are we going ahead with this? Can we push this towards merging? What's needed?

goodov commented:
> Are we going ahead with this? Can we push this towards merging? What's needed?

@wknapik https://github.com/brave/devops/issues/15827

thypon commented:
Closed as superseded by brave/brave-browser#56286.
